cdap-api/src/main/java/co/cask/cdap/api/lineage/LineageGraph.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.lineage;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Graph representing set of {@link Read} operations from the data sources,
+ * set of {@link Transform} operations, and set of {@link Write} operations to the data sinks.
+ */
+public class LineageGraph {
+  private final Set<Read> reads;
+  private final Set<Transform> transforms;
```

Comment:
I don't think this is a set, because the order of operations matters.

Reply:
The order is implicit here from the `from` and `to` constructs. For example lets say you have 3 transformation  (A->B), (B->C), (C->D). It does not matter in what order you add them in the collection {(A->B), (B->C), (C->D)} or {(B->C), (A->B), (C->D)}, you always get the lineage as A->B->C->D

Reply:
Say you have two operations:

- a = a + b;
- c = a + b

I would think the order matters, because depending on the order, c ends up being either a + b or a + b + b. You can construct similar case with dropping a field, then renaming another field to the same name that was dropped. Or Date normalization, is after adjusting seonds to milliseconds by multiplying with 1000, or after? etc.